onnx/backend/test/runner/__init__.py

```diff
@@ -257,10 +263,18 @@ def run(test_self, device):
                         tensor.ParseFromString(f.read())
                     ref_outputs.append(numpy_helper.to_array(tensor))
                 outputs = list(prepared_model.run(inputs))
-                self._assert_similar_outputs(ref_outputs, outputs)
+                self._assert_similar_outputs(ref_outputs, outputs, rtol, atol)
 
         self._add_test(kind + 'Model', model_test.name, run, model_marker)
 
+    def _tolerance(self, name):
+        for pattern, rtol, atol in self._tolerance_patterns:
```

Comment:
You can finalize these configurations in `tests_cases`, `test_suites` and `tests` functions, to avoid doing this search repeatedly inside the real tests (extend `Testitem` store the corresponding `rtol` and `atol` as metadata).